dspy/predict/predict.py

```diff
@@ -108,6 +108,17 @@ def _forward_preprocess(self, **kwargs):
         if (temperature is None or temperature <= 0.15) and num_generations > 1:
             config["temperature"] = 0.7
 
+        if "prediction" in kwargs:
+            if (
```

Comment:
Happy to merge for now but:

1) Should we just check (or maybe _also_ check?) whether `prediction` is a key in the signature?

2) I suppose the cleanest way for this would have been to make it a new type and handle it in adapters 😅  But this would start to feel like too many types...